src/core/acroform.cpp

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: MPL-2.0
+
+#include <qpdf/Constants.h>
+#include <qpdf/Types.h>
+#include <qpdf/DLL.h>
+#include <qpdf/QPDFExc.hh>
+#include <qpdf/QPDFFormFieldObjectHelper.hh>
+#include <qpdf/QPDFAcroFormDocumentHelper.hh>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "pikepdf.h"
+
+void init_acroform(py::module_ &m)
+{
+    py::class_<QPDFFormFieldObjectHelper,
+        std::shared_ptr<QPDFFormFieldObjectHelper>,
+        QPDFObjectHelper>(m, "FormField")
+        .def(py::init<QPDFObjectHandle &>(), py::keep_alive<0, 1>())
+        .def_property_readonly("is_null", &QPDFFormFieldObjectHelper::isNull)
+        .def_property_readonly("parent", &QPDFFormFieldObjectHelper::getParent)
+        .def_property_readonly("top_level_field",
+            [](QPDFFormFieldObjectHelper &field) { return field.getTopLevelField(); })
+        .def("get_inheritable_field_value",
+            &QPDFFormFieldObjectHelper::getInheritableFieldValue,
+            py::arg("name"))
+        .def("get_inheritable_field_value_as_string",
+            &QPDFFormFieldObjectHelper::getInheritableFieldValueAsString,
+            py::arg("name"))
+        .def("get_inheritable_field_value_as_name",
+            &QPDFFormFieldObjectHelper::getInheritableFieldValueAsName,
+            py::arg("name"))
+        .def_property_readonly("field_type", &QPDFFormFieldObjectHelper::getFieldType)
+        .def_property_readonly(
+            "fully_qualified_name", &QPDFFormFieldObjectHelper::getFullyQualifiedName)
+        .def_property_readonly(
+            "partial_name", &QPDFFormFieldObjectHelper::getPartialName)
+        .def_property_readonly(
+            "alternate_name", &QPDFFormFieldObjectHelper::getAlternativeName)
+        .def_property_readonly(
+            "mapping_name", &QPDFFormFieldObjectHelper::getMappingName)
+        .def_property_readonly("value", &QPDFFormFieldObjectHelper::getValue)
+        .def_property_readonly(
+            "value_as_string", &QPDFFormFieldObjectHelper::getValueAsString)
+        .def_property_readonly(
+            "default_value", &QPDFFormFieldObjectHelper::getDefaultValue)
+        .def_property_readonly("default_value_as_string",
+            &QPDFFormFieldObjectHelper::getDefaultValueAsString)
+        .def_property_readonly(
+            "default_appearance", &QPDFFormFieldObjectHelper::getDefaultAppearance)
+        .def_property_readonly(
+            "default_resources", &QPDFFormFieldObjectHelper::getDefaultResources)
+        .def_property_readonly("quadding", &QPDFFormFieldObjectHelper::getQuadding)
+        .def_property_readonly("flags", &QPDFFormFieldObjectHelper::getFlags)
+        .def_property_readonly("is_text", &QPDFFormFieldObjectHelper::isText)
+        .def_property_readonly("is_checkbox", &QPDFFormFieldObjectHelper::isCheckbox)
+        // An `isChecked` method is also documented in the header, but doesn't seem to actually be implemented
```

Comment:
You're right; I think this was just missed on the qpdf side.
I opened a PR to fix that issue. pikepdf would be better off assuming the API is not available yet and using the implementation I provided.

https://github.com/qpdf/qpdf/pull/1439